modules/syslogformat/tests/test_fuzzing_syslogformat.c

```diff
@@ -0,0 +1,35 @@
+
+#include <stdint.h>
+#include <stddef.h>
+
+#include "apphook.h"
+#include "msg-format.h"
+#include "logmsg/logmsg.h"
+#include "cfg.h"
+#include "syslog-format.h"
+#include <iv.h>
+
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+  if (size <= 1) return 0;
+
+  GlobalConfig *cfg = cfg_new_snippet();
+  app_startup();
+  cfg_load_module(cfg, "syslogformat");
+  MsgFormatOptions parse_options;
+  msg_format_options_defaults(&parse_options);
+
+  LogMessage *msg = log_msg_new_empty();
+
+  syslog_format_handler(&parse_options, data, size, msg);
+
+  log_msg_unref(msg);
+  cfg_free(cfg);
+  app_shutdown();
+  iv_deinit();
```

Comment:
Why do we need this `iv_deinit`?

Reply:
I do not recall if it was a crash or simply memory leak. But neither is acceptable for fuzzing.
Actually it should be called by `app_shutdown`, but it is not
https://github.com/syslog-ng/syslog-ng/blob/787f588d2ac6d91a40b3289e8857e2487586efb7/lib/apphook.c#L228-L238

Yes, most of our UT (where `app_startup` is used) should call `iv_deinit` but we can get away with it, as it is not called multiple times (sandboxing the tests "helps").